src/core/thread/dua_manager.cpp

```diff
@@ -66,33 +68,110 @@ void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefix
         Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
```

Comment:
Should set `mDomainUnicastAddress .mValid = false` ? 
If not, after `DUA` is removed from `Netif`, `DuaManager::GetDomainUnicastAddress()` will still return the "DUA" that is not effective.